### PR TITLE
Add indexing to CLI

### PR DESCRIPTION
**Pull request by JJ-Pineda**
Added actions for indexing to the command line interface: create-index, delete-index, and bulk-index (from json file of smiles and extra_data).

### CODE REVIEW

File: src/command_line/indexing/bulk_index.rs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+use tantivy::schema::Field;
+
+pub const NAME: &str = "bulk-index";
+
+pub fn command() -> Command {
+    Command::new(NAME)
+        .arg(
+            Arg::new("index_path")
+                .required(true)
+                .long("index_path")
```
Comment (xrl) [inline src/command_line/indexing/bulk_index.rs:18]: I think the convention is dashed, not underscored, could you try `index-path`?

File: src/command_line/indexing/bulk_index.rs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    let reader = BufReader::new(file);
+    let json_array: Vec<Value> = serde_json::from_reader(reader)?;
+
+    let _ = json_array
+        .into_par_iter()
+        .map(|ob| {
+            let smile = ob
+                .get("smile")
+                .ok_or(eyre::eyre!("Failed to extract smiles"))
+                .unwrap()
```
Comment (xrl) [inline src/command_line/indexing/bulk_index.rs:60]: Sure would be nice if `?` worked in closures! The unwrap() is fine by me.

File: src/command_line/indexing/create_index.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+use crate::command_line::indexing::split_path;
+use crate::indexing::index_manager::IndexManager;
+use std::ops::Deref;
+
+pub const NAME: &str = "create-index";
+
+pub fn command() -> Command {
+    Command::new(NAME)
+        .arg(
+            Arg::new("index_path")
```
Comment (xrl) [inline src/command_line/indexing/create_index.rs:11]: I think dashes are good here. But again, please confirm the other bits of the codebase use dashes 😆

File: src/command_line/indexing/mod.rs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        .ok_or(eyre::eyre!("Could not extract storage directory"))?
+        .to_str()
+        .ok_or(eyre::eyre!("Could not convert storage director to str"))?
+        .to_string();
+    let child_name = index_path_path
+        .file_stem()
+        .ok_or(eyre::eyre!("Could not extract index name"))?
+        .to_str()
+        .ok_or(eyre::eyre!("Could not convert index name to str"))?
+        .to_string();
```
Comment (xrl) [inline src/command_line/indexing/mod.rs:26]: Idiomatic as heck over here!

File: src/command_line/search/basic_search.rs [hunk 5]
```diff
@@ -1,8 +1,8 @@
-pub use super::prelude::*;
+pub use super::super::prelude::*;
```
Comment (xrl) [inline src/command_line/search/basic_search.rs:1]: I wonder if we should make `use` statements relative to `crate`? Mixing and matching relative vs absolute could be confusing.

File: src/rest_api/openapi_server.rs [hunk 6]
```diff
@@ -105,7 +98,7 @@ impl Api {
 
     #[oai(path = "/v1/indexes/:index", method = "get")]
     /// Get extended information about an index
-    pub async fn v1_get_index(&self, index: Path<String>) -> GetIndexesResponse {
+    pub async fn v1_get_index(&self, index: Path<String>) -> GetIndexResponse {
```
Comment (xrl) [inline src/rest_api/openapi_server.rs:101]: Did I really have the wrong response type for `v1_get_index`? 🤦

File: src/search/mod.rs [hunk 7]
```diff
@@ -95,7 +103,7 @@ pub fn aggregate_query_hits(
             get_smiles_and_extra_data(result, &searcher, smile_field, extra_data_field)?;
 
         final_results.push(QuerySearchHit {
-            extra_data: extra_data.into(),
+            extra_data: serde_json::from_str(extra_data.deref())?,
```
Comment (xrl) [inline src/search/mod.rs:106]: `extra_data` is type `String`, right? I think it's more idiomatic to do `&extra_data` and I believe rust will make it in to an `&str`.

**Comment by xrl** [comment]
@JJ-Pineda is this good to merge?

**Comment by JJ-Pineda** [comment]
The only thing that's missing is parallelized doc writing for the `bulk-index` option. Chunks didn't seem to work with the bufreader for some reason. Almost there...

**Review by xrl — APPROVED** [review]

**Pull request merged**